homeassistant/components/weatherflow/sensor.py

```diff
@@ -0,0 +1,417 @@
+        self.device = device
+        self.entity_description = description
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.device.serial_number)},
+            manufacturer="WeatherFlow",
+            model=self.device.model,
+            name=f"{self.device.model} {self.device.serial_number}",
+            sw_version=self.device.firmware_revision,
+            suggested_area="Backyard",
+        )
+
+        self._attr_unique_id = f"{self.device.serial_number}_{description.key}"
+
+        # In the case of the USA - we may want to have a suggested US unit which differs from the internal suggested units
+        if (description.imperial_suggested_unit is not None) and (not is_metric):
+            self._attr_suggested_unit_of_measurement = (
+                description.imperial_suggested_unit
+            )
+
+    @property
+    def last_reset(self) -> datetime | None:
+        """Return the time when the sensor was last reset, if any."""
+        if self.entity_description.state_class == SensorStateClass.TOTAL:
+            return self.device.last_report
+        return None
+
+    @property
+    def native_value(self) -> datetime | StateType:
+        """Return the state of the sensor."""
+
+        # Extract raw sensor data
+        # Either pull from the key (default) or (backing_library_attribute) to get sensor value
+        if self.entity_description.backing_library_attribute_fn() != "":
+            raw_sensor_data = getattr(
+                self.device, self.entity_description.backing_library_attribute_fn()
+            )
+        else:
+            raw_sensor_data = getattr(self.device, self.entity_description.key)
+
+        normalized_data = raw_sensor_data
+
+        if isinstance(normalized_data, Quantity):
+            sensor_value = normalized_data.magnitude
+            return sensor_value
+        if isinstance(normalized_data, Enum):
+            sensor_value = normalized_data.name
+            return sensor_value
+        if isinstance(normalized_data, float):
+            return normalized_data
+        if isinstance(normalized_data, int):
+            return normalized_data
+        return None
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to events."""
+        for event in self.entity_description.event_subscriptions:
+            self.async_on_remove(
+                self.device.on(event, lambda _: self.async_write_ha_state())
+            )
+
+
+class WeatherFlowAirDensitySensorEntity(WeatherFlowSensorEntity):
+    """Special case where we have a custom function."""
+
+    entity_description: AirDensityWeatherFlowSensorEntityDescription
+
+    def __init__(
+        self,
+        device: WeatherFlowDevice,
+        description: WeatherFlowSensorEntityDescription,
+        is_metric: bool = True,
+    ) -> None:
+        """Initialize the class."""
+        super().__init__(device, description, is_metric)
+        self.is_metric = is_metric
```

Comment:
You don't use is_metric

Reply:
![image](https://github.com/home-assistant/core/assets/6491743/c3d8c2b0-08e9-44b4-9125-169089cab65b)

Reply:
hehe, you ctrl+F'd the file. You use is_metric at least somewhere, but not at this place :)

Reply:
Ok I realized what the issue here is.

In App we are either using "lb/ft3" or "kg/m3" neither of which are supported in the HA Device class.... i'll just use the intenral options :)